src/components/CommentCard.jsx

```diff
@@ -1,10 +1,49 @@
-import { Link } from "react-router-dom"
+import { Link } from "react-router-dom";
+import { useState } from "react";
 import { relativeTimeFormatter } from "../../utils/formatter-utils";
+import apiRequest from "../../utils/api-utils";
+
+const CommentCard = ({ comment: { body, created_at, author, votes: currVotes, comment_id, article_id } }) => {
+    const [votes, setVotes] = useState(currVotes);
+
+    const voteHandler = (num) => ({ target }) => {
+        const oppositeBtn = [...target.parentElement.children].find(field => field.nodeName === "BUTTON" && field !== target);
+
+        const lockBtns = (bool) => [target, oppositeBtn].forEach((btn) => {
+            btn.classList[bool ? "add" : "remove"]("btn-lock");
+            btn.classList.remove("btn-error");
+        });
+
+        lockBtns(true);
+
+        const relativeUrlPath = comment_id == null ? `articles/${article_id}` : `comments/${comment_id}`;
+        apiRequest("patch", relativeUrlPath, { inc_votes: num })
+            .then(({ data }) => {
+                setVotes((data.article ?? data.comment)["votes"]);
+
+                if (oppositeBtn.classList.contains("btn-disabled"))
+                    [target, oppositeBtn].forEach((btn) => btn.classList.remove("btn-disabled"));
+                else
+                    target.classList.add("btn-disabled");
+
+                lockBtns(false);
+            })
+            .catch(() => {
+                lockBtns(false);
+                target.classList.add("btn-error");
+            });
+    };
 
-const CommentCard = ({ comment: { body, created_at, author } }) => {
     return (
         <div className="comment-card">
-            <p>{body}</p>
+            <div className="comment">
+                <div className="voting">
+                    <button className="round" onClick={voteHandler(1)}>&#x25B3;</button>
```

Comment:
onClick should be given an arrow function or a function reference as its value, so it can call the function when a user clicks the button